### PR TITLE
Hg 834 Componentizing the top nav

### PR DESCRIPTION
**Pull request by lizlux**
We want to be able to share the "global nav" between applications, so I've created it as a web component inside the style guide. There will be a parallel PR for bringing it into the mercury repo, although that will be a WIP. 

Note, I updated gulp-sass and node-sass in order to fix an bug with error reporting, however this has had the unintended consequence of altering a number of css files :(. I tried to revert back to the original versions but couldn't find a way to do it without getting a bunch of errors. For code reviewers: The first file that I edited after all the css changes is `gh-pages/_includes/head.html`, so search for that string to bypass the mayhem.  

Some to do's: 
- [x] try using the `-lite` version of the web components library
- [x] update `global-nav.html` to include new `tob-bar` custom tag
- [x] add class of `.global-navigation` to `top-bar` invocation to keep stylesheet the same.

### CODE REVIEW

File: dist/components/top-bar/top-bar.html [hunk 1]
```diff
@@ -0,0 +1,12 @@
...
+			<div class="replace"></div>
+		</div>
+	</div>
+</template>
+
+<script>!function(e,t){"use strict";e.SG=e.SG||{},SG.Components=SG.Components||{};var o=SG.Components.TopBar=function(t){this.doc=t.importedDoc||e.document,this.logoUrl=t.logoUrl||"",this.logoHref=t.logoHref||"/",this.elem=null};o.prototype.createElement=function(){var t=Object.create(HTMLElement.prototype),o=this;t.createdCallback=function(){var t=o.doc!==e.document,r=Array.prototype.slice.call(this.children),n=o.doc.getElementById("SGTopBarTemplate").content,i=t?o.doc.importNode(n,!0):n,l=i.querySelector(".replace");this.innerHTML="",o.updateLogo(i),r.forEach(function(e){l.parentElement.insertBefore(e,l)}),this.appendChild(i),this.style.visibility="visible"},this.elem=e.document.registerElement("top-bar",{prototype:t})},o.prototype.updateLogo=function(e){var t=e.querySelector(".logo");t.src=this.logoUrl,t.parentElement.href=this.logoHref}}(window,document);</script>
+<style>top-bar{display:block;visibility:hidden;background-color:#fff;border-bottom:1px solid #ccc;height:46px;position:fixed;width:100%;z-index:9999999}top-bar .logo{fill:#092344;width:82px}
+       </style>
```
Comment (lizlux) [inline dist/components/top-bar/top-bar.html]: This is the actual component file that gets imported into your app.

File: dist/css/lib/components/block-grid.css [hunk 2]
```diff
@@ -27,7 +27,9 @@ meta.foundation-data-attribute-namespace {
 html, body {
```
Comment (lizlux) [inline dist/css/lib/components/block-grid.css:27]: You can ignore all of this until `gh-pages/_includes/head.html`

File: gh-pages/vendor/wikia-style-guide/dist/css/lib/components/block-grid.css [hunk 3]
```diff
@@ -27,7 +27,9 @@ meta.foundation-data-attribute-namespace {
 html, body {
```
Comment (lizlux) [inline gh-pages/vendor/wikia-style-guide/dist/css/lib/components/block-grid.css:27]: You can ignore the css changes till `gh-pages/vendor/wikia-style-guide/dist/vendor/webcomponentsjs/CustomElements.js`, which is a library upgrade, and then pick back up at `gulpfile.js`.

**Comment by kenkouot** [comment]
So far we are only using the webcomponents set of polyfills. I wonder if we should be using something slightly higher level. For instance if we use Polymer, we gain access to Vulcanize.
https://github.com/polymer/vulcanize

Or we could use SkateJS as a loosely opinionated Custom Elements library that is significantly lighter in filesize than the other options.

**Comment by kenkouot** [comment]
This PR is also really hard to read, how do you feel about regenerating /dist files in another PR, only when we bump releases?

File: src/components/top-bar/top-bar.js [hunk 4]
```diff
@@ -0,0 +1,61 @@
+// todo: use ES6 instead of namespace
+(function (window, document) {
+	'use strict';
+
+	window.SG = window.SG || {};
```
Comment (kenkouot) [inline src/components/top-bar/top-bar.js:5]: This kind of boilerplate is not necessary. Let's just write a namespacing function and make a core styleguide js file for common functions/utility like this.

**Comment by lizlux** [comment]
closing this in favor of https://github.com/Wikia/style-guide/pull/56/files

**Comment by lizlux** [comment]
Closing this in favor of https://github.com/Wikia/style-guide/pull/56

File: src/components/top-bar/top-bar.js (hunk 4 above)
Comment (lizlux) [inline src/components/top-bar/top-bar.js:5]: Discussed handling this in a separate commit, possibly with Polymer.

File: gh-pages/_layouts/default.html [hunk 5]
```diff
@@ -12,6 +12,16 @@
       </div>
     </div>
 
+    <script>
+      document.addEventListener('DOMContentLoaded', function () {
+        new window.SG.Components.TopBar({
```
Comment (kenkouot) [inline gh-pages/_layouts/default.html:17]: I'm not a huge fan of this sort of manual invocation. It's just boilerplate and variables like logoUrl and logoHref should just be element attributes. Going to look into a different approach for next iteration.

**Comment by kenkouot** [comment]
Merging for now, will address improvements in V2.

**Pull request merged**